sys-kernel/linux-headers/linux-headers-3.18-r1.ebuild

```diff
@@ -0,0 +1,55 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+ETYPE="headers"
+H_SUPPORTEDARCH="alpha amd64 arc arm arm64 avr32 bfin cris frv hexagon hppa ia64 m32r m68k metag microblaze mips mn10300 openrisc ppc ppc64 s390 score sh sparc tile x86 xtensa"
+inherit kernel-2
+detect_version
+
+PATCH_VER="1"
+SRC_URI="mirror://gentoo/gentoo-headers-base-${PV}.tar.xz
+	${PATCH_VER:+mirror://gentoo/gentoo-headers-${PV}-${PATCH_VER}.tar.xz}"
+
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-linux ~arm-linux ~x86-linux"
+
+DEPEND="app-arch/xz-utils
+	dev-lang/perl"
+RDEPEND="!!media-sound/alsa-headers"
+
+S=${WORKDIR}/gentoo-headers-base-${PV}
+
+src_unpack() {
+	unpack ${A}
+}
+
+src_prepare() {
+	default
+
+	[[ -n ${PATCH_VER} ]] && EPATCH_SUFFIX="patch" epatch "${WORKDIR}"/${PV}
```

Comment:
Why `epatch` and not `eapply`?

Reply:
because I'm not going that deep into it and its what's used on all the other ebuilds.
I can alter that if you like, however.

Reply:
If this change is not moving `sys-kernel/linux-headers` to `EAPI=6` then please elaborate whan exactly you are trying to achieve in the commit message itself.

Reply:
Ah. The major point is moving away from lower EAPI to free up the reqs for
kernel-2.eclass, but I can go ahead and do it for all of the linux-headers ebuilds.

Reply:
@trofi went ahead and added EAPI 6 ebuilds, and migrated the only EAPI 6
one from epatch to eapply.